crates/egui/src/widgets/image.rs

```diff
@@ -288,8 +288,24 @@ impl<'a> Image<'a> {
     }
 
     #[inline]
-    pub fn source(&self) -> &ImageSource<'a> {
-        &self.source
+    pub fn source(&'a self, ctx: &Context) -> ImageSource<'a> {
+        match &self.source {
+            ImageSource::Bytes { uri, .. } => match uri.starts_with("gif://") {
+                true => Some(uri),
+                false => None,
+            },
+            _ => None,
```

Comment:
We should be able to handle `ImageSource::Uri` here too, for showing images from e.g. a HTTP url (but we don't need to squeeze it in to this PR)